### PR TITLE
[chore] multimod must be run before genotelcorecol

### PR DESCRIPTION
**Pull request by codeboten**
Adding a step in `prepare-release` to merge the branch created by multimod.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -407,15 +407,16 @@ endif
...
 	git checkout -b opentelemetry-collector-bot/release-$(RELEASE_CANDIDATE)
 	git add .
 	git commit -m "prepare release $(RELEASE_CANDIDATE)"
 	$(MAKE) multimod-prerelease
-	# commit multimod changes
+	# merge multimod changes
+	git merge prerelease_collector-core_v$(RELEASE_CANDIDATE)
+	# regenerate files
+	$(MAKE) -C cmd/builder config
+	$(MAKE) genotelcorecol
```
Comment (bogdandrutu) [inline Makefile:419]: need to delete the branch?

Comment (bogdandrutu) [reply Makefile:419]: Also, what about https://github.com/open-telemetry/opentelemetry-go-build-tools/pull/134 ?

Comment (codeboten) [reply Makefile:419]: having multimod not require a new branch would be great. i'll review that change tomorrow 👍🏻

Comment (bogdandrutu) [reply Makefile:419]: Ping?

Comment (bogdandrutu) [reply Makefile:419]: PR was merged, consider to bump version and use that.

Comment (codeboten) [reply Makefile]: Updated to use latest multimod w/ `-b=false` set to prevent it from creating a separate branch

File: internal/tools/go.mod [hunk 2]
```diff
@@ -12,8 +12,8 @@ require (
 	github.com/tcnksm/ghr v0.15.0
 	github.com/wadey/gocovmerge v0.0.0-20160331181800-b5bfa59ec0ad
 	go.opentelemetry.io/build-tools/checkdoc v0.0.0-20220327151007-0b1d24fa04f4
```
Comment (bogdandrutu) [inline internal/tools/go.mod:14]: Update this as well?

**Review by bogdandrutu — APPROVED** [review]

File: Makefile [hunk 3]
```diff
@@ -379,7 +379,7 @@ multimod-verify: install-tools
 
 .PHONY: multimod-prerelease
 multimod-prerelease: install-tools
-	multimod prerelease -v ./versions.yaml -m collector-core
+	multimod prerelease -b=false -v ./versions.yaml -m collector-core
```
Comment (bogdandrutu) [inline Makefile:382]: "-s=true" to skip `go mod tidy` from the tool?

https://github.com/open-telemetry/opentelemetry-go-build-tools/blob/main/multimod/cmd/prerelease.go#L83

**Review by bogdandrutu — APPROVED** [review]

File: Makefile (hunk 3 above)
Comment (codeboten) [reply Makefile]: 👍🏻 done

File: internal/tools/go.mod (hunk 2 above)
Comment (codeboten) [reply internal/tools/go.mod]: Done

**Pull request merged**